core/src/main/java/com/datastax/oss/driver/api/core/metadata/Node.java

```diff
@@ -157,10 +178,10 @@
    * The host ID that is assigned to this node by Cassandra. This value can be used to uniquely
    * identify a node even when the underling IP address changes.
    *
-   * <p>This should be non-null in a healthy deployment, but the driver will still function, and
-   * report {@code null} here, if the server metadata was corrupted.
+   * <p>This information is required; the driver will not function properly if this method returns
+   * {@code null}.
    */
-  @Nullable
+  @NonNull
   UUID getHostId();
```

Comment:
So I double-checked this and this needs to be non-null, otherwise the driver will throw NPEs in many places. I changed the peer check logic in `DefaultTopologyMonitor` to include the broadcast RPC address and the host ID.

Reply:
My bad, it only really became non-null after [JAVA-2165](https://datastax-oss.atlassian.net/browse/JAVA-2165), I forgot to change it then.